src/dynarec/la64/dynarec_la64_emit_shift.c

```diff
@@ -444,3 +444,58 @@ void emit_ror32c(dynarec_la64_t* dyn, int ninst, rex_t rex, int s1, uint32_t c,
         }
     }
 }
+
+// emit ROL32 instruction, from s1, s2, store result in s1 using s3 and s4 as scratch
+void emit_rol32(dynarec_la64_t* dyn, int ninst, rex_t rex, int s1, int s2, int s3, int s4)
+{
+    int64_t j64;
+
+    IFX (X_PEND) {
+        SDxw(s2, xEmu, offsetof(x64emu_t, op2));
+        SET_DF(s4, rex.w ? d_rol64 : d_rol32);
+    } else IFX (X_ALL) {
+        SET_DFNONE();
+    }
+
+    if (la64_lbt) {
+        IFX (X_ALL) {
+            if (rex.w)
+                X64_ROTL_D(s1, s2);
+            else
+                X64_ROTL_W(s1, s2);
+        }
+    }
```

Comment:
missing an else of something? that's strange s1 get rol'd if la64_lbt and IFX, and rol'd again before line 485 were it finaly return.

Reply:
> Or is that just flags update and not the value itself?

Yes, LBT instructions prefixed with `X64_` only update flags. The design philosophy of LBT is to keep the RISC instructions RISCy and add additional x86-related simulations on top of it.